design/thin-edge-core.md

```diff
@@ -0,0 +1,50 @@
+# Thin-edge Core Vision
+
+A typical IoT application has to address:
+
+* __sophisticated business-cases__, with applications involving several vendors, actors and users;
+* __very diverse integration needs__ over operating systems, protocols, data sources, processing tools, actuators, cloud end-points;
+* __constraint environments__ with low-resources devices, large-scale deployments, high exposure to security risks,
+  prohibitive manual operations, long-term supports;
+* __safety related requirements__ with devices operating machines in hazardous contexts.
+  
+The forces being these requirements are pushing in opposite directions.
+
+* On one side, one needs a dynamic and open setting with a large set of features that can be easily extended and combined.
+  [NodeRed](https://nodered.org/) is the archetype of such tools facilitating connections of components.
+* On the other side, one needs a minimal system specifically built for a task with cherry-picked components.
```

Comment:
I feel this requirement is not pushing in the opposite direction, but supportive for `__constraint environments__ `.  This will allow thin-edge.io to be built and run on resource-constrained devices. We just have to facilitate/be flexible enough to pick/choose and build the components that suit the use case.

Reply:
What I meant is that what is supporting the environment constraints are pushing in a direction that is opposite to flexibility and generality.